packages/back-end/src/util/secrets.ts

```diff
@@ -173,3 +173,35 @@ export const SECRET_API_KEY = secretAPIKey;
 export const PROXY_ENABLED = !!process.env.PROXY_ENABLED;
 export const PROXY_HOST_INTERNAL = process.env.PROXY_HOST_INTERNAL || "";
 export const PROXY_HOST_PUBLIC = process.env.PROXY_HOST_PUBLIC || "";
+
+/**
+ * Allows custom configuration of the trust proxy settings as
+ * described in the docs: https://expressjs.com/en/5x/api.html#trust.proxy.options.table
+ *
+ * Supports true/false for boolean values.
+ * Supports integer values to trust the nth hop from the front-facing proxy server as the client.
+ * All other truthy values will be used verbatim.
+ */
+const getTrustProxyConfig = (): boolean | string | number => {
+  const value = process.env.EXPRESS_TRUST_PROXY_OPTS;
+
+  // If no value set, return false
+  if (!value) {
+    return false;
+  }
+
+  // Lower-cased value to enable easier boolean config
+  const lowerCasedValue = value.toLowerCase();
+  if (lowerCasedValue === "true") return true;
+  if (lowerCasedValue === "false") return false;
+
+  // Check for nth hop config
+  //    Trust the nth hop from the front-facing proxy server as the client.
+  const maybeNumber = parseInt(value);
```

Comment:
This won't work since ips are valid strings people might use
```ts
parseInt("127.0.0.1") === 127 // true
```

Since we only support integers, a regex might be better:
```ts
if (value.match(/^[0-9]+$/)) {
  ...
}
```

Reply:
Updated in [299f954](https://github.com/growthbook/growthbook/pull/1149/commits/299f954ae07c46898227117e00d24f3fc27a7d94)